client/src/client/thunks/thunks.js

```diff
@@ -2,15 +2,15 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
 import { actionTypes } from '../actionTypes/actionTypes';
 import Service from '../services/service';
 
-export const getItemsAsync = createAsyncThunk(
+export const getSessionsAsync = createAsyncThunk(
   actionTypes.GET_SESSIONS,
   async () => {
     return await Service.getSessions();
   }
 );
-export const filterItemsAsync = createAsyncThunk(
+export const filterSessionsAsync = createAsyncThunk(
   actionTypes.FILTER_SESSIONS,
-  async (item) => {
-    return await Service.filterSessions({ item });
+  async (filter) => {
+    return await Service.filterSessions({ filter });
```

Comment:
I used these methods mainly because the way instructor explained in the lecture - that it is the best way to handle the states of fetching data . If you will make this sync, how will you handle the `pending` state i.e. when the data is being fetched? Have you tried it? - I will also look more into it. 
Can you give me an example of the use cases of synchronous behaviour? 
Also, I don't understand what do you mean by regular thunks?

Reply:
I used the regular version of them for my assignment and they were a lot more straightforward for me than using the `createAsyncThunk` function. At the top of the page in the URL above, you can see some examples of those.

For example, we can keep all of our existing code the same and just include another function like such to handle the API calls:

```
function complexSynchronousThunk(someValue) {
  return (dispatch, actionName) => {
    res = someAPIcall(someValue)
    dispatch(actionName(res))
  }
}
```

Basically, they handle calling APIs by just making the dispatch call async after we get a result from an endpoint call. Because it's just handled through base JS promises, it lets just customize behaviour. In cases where we need extra states besides pending we're free to add whatever, and in cases where we don't need any of that we can keep our code very simple with just a promised dispatch. 

Honestly, the flexibility of the code and lack of a complicated abstraction just makes it seem easier to work with. Also idk if the instructor chose to demo the `createAsyncThunk` function because its easier, rather probably because that's what she was familiar with from when she worked with thunks "years ago" lmao.